src/NEO/Address.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+
+#include "../Hash.h"
+
+#include <TrezorCrypto/base58.h>
+
+using namespace TW::NEO;
+
+bool Address::isValid(const std::string& string) {
+    size_t capacity = 128;
+    uint8_t buffer[capacity];
+
+    int size = base58_decode_check(string.data(), HASHER_SHA2D, buffer, (int)capacity);
+    if (size != Address::size || buffer[0] != version) {
+        return false;
+    }
+
+    return true;
+}
+
+Address::Address(const std::string& string) {
```

Comment:
This looks like a copy of Base58 address chain, did you consider reusing code from other chains?

Reply:
Yes, I'm going to use `std::string bytesToBase58(const uint8_t *data, size_t dataSize)` from Tezos

Reply:
@alejandro-isaza can you advise how should we move forward with this? there is only one byte that is different, any way to derivate base class?

Reply:
Yes, we should definitely refactor. I can add this to my todo list.

Reply:
done